### PR TITLE
Add cw20 usage documentation

### PR DESCRIPTION
**Pull request by ethanfrey**
In this section, we fire up `@cosmjs/cli` and go deep with the cw20 contract.

Please check it out (also with `yarn serve`). I would like to publish this to all testnet users Friday (today). And we can link it from a medium article.

Let me know if anything doesn't make sense... I stayed up too late working on this, it probably gets a bit hard to follow.

### CODE REVIEW

File: learn/using-contracts/cw20.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+```ts
+const client = await useOptions(coralnetOptions).setup(YOUR_PASSWORD_HERE);
+
+const cw20 = CW20(client);
+const codeId = await cw20.upload();
+console.log(`CodeId: ${codeId}`);
+// output: 4
+
+.editor
```
Comment (orkunkl) [inline learn/using-contracts/cw20.md:36]:
```suggestion
// enable REPL editor mode to edit multiline code then execute
.editor
```

File: learn/using-contracts/cw20.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    { address: "coral1qvrcashqpemlkhrqphzv9n5nutdxpafmdefgcl", amount: "10000"},
+    { address: "coral14f8nvyy4c9pyn78dgv0k6syek3jjjrkyz747kj", amount: "10000"},
+    { address: "coral1e86v774dch5uwkks0cepw8mdz8a9flhhapvf6w", amount: "10000"},
+    { address: "coral1hf50trj7plz2sd8cmcvn7c8ruh3tjhc2nhyl7l", amount: "10000"},
+  ],
+  mint: {
+    minter: client.senderAddress,
+  },
+};
+^D
```
Comment (orkunkl) [inline learn/using-contracts/cw20.md:56]:
```suggestion
// exit editor using `^D` and execute entered code 
^D
```

File: learn/using-contracts/cw20.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+  increaseAllowance: [AsyncFunction: increaseAllowance],
+  decreaseAllowance: [AsyncFunction: decreaseAllowance],
+  transferFrom: [AsyncFunction: transferFrom] }
+```
+
+But how do I call them? What arguments do they take?
+There is a great `.type` operator. Unfortunately it doesn't work
+with members directly, but with a local variable, we can get the info.
+
+This is not what we want:
```
Comment (orkunkl) [inline learn/using-contracts/cw20.md]: Telling negative results before showing code is a bit confusing. The reader will be like: **then why do you show this to me?**. Swap the lines of the code example below and this line.

File: learn/using-contracts/cw20.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+const contractAddress = "coral13nkgqrfymug724h8pprpexqj9h629sa3yzcqfk"
+
+// now, connect to that contract and make sure it is yours
+const mine = cw20.use(contractAddress);
+mine.token_info()
+mine.minter()
+mine.balance()
+```
+
+Okay, you are connected to your contract. Let's DO SOMETHING.
```
Comment (orkunkl) [inline learn/using-contracts/cw20.md]:
```suggestion
Okay, you are connected to your contract. Let's see what cw20 is capable of.
```

File: learn/using-contracts/setup.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+work properly on Windows (we assume a HOME environmental variable). PRs welcome.
+
+## Connecting to the chain
+
+The first step before doing anything is ensuring we can create an account and connect to the chain.
+You will always use the following command to start up the `@cosmjs/cli` with some cw20-specific helpers preloaded
+(in addition to all the general helpers it has).
+
+```shell
+npx @cosmjs/cli --init https://raw.githubusercontent.com/CosmWasm/cosmwasm-plus/master/contracts/cw20-base/helpers.ts
```
Comment (orkunkl) [inline learn/using-contracts/setup.md:30]: I really like this `helpers` existence. Very useful. Maybe we can add embed contract helpers URL inside tx msg like source and builder. And in the wasm.glass as well 
image

**Review by orkunkl — APPROVED** [review]
Looks good. Just a few comments

File: learn/using-contracts/cw20.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+```
+
+If you want to see this is real, just check out
+[Code ID 4](https://coralnet.wasm.glass/#/codes/4)
+and [contract `coral16t7y0vrtpqjw2d7jvc2209yan9002339mg4mrv`](https://coralnet.wasm.glass/#/contracts/coral16t7y0vrtpqjw2d7jvc2209yan9002339mg4mrv)
+on the wasm.glass code explorer.
+
+### Try this at home: MINE
+
+Now that we have that uploaded, we can easily make a second contract. This one, please
```
Comment (orkunkl) [inline learn/using-contracts/cw20.md:75]: I found great vuepress gems today. https://v1.vuepress.vuejs.org/guide/markdown.html#custom-containers. We can use warning, tip containers to point out warning. I'll go through the documentation after this PR and add containers.

File: learn/using-contracts/cw20.md (hunk 3 above)
Comment (ethanfrey) [reply learn/using-contracts/cw20.md]: Good idea. After doing the proper way, you will think "why so long, this is a shortcut".
I tried the first one first, but yeah, I guess you want the answer first

File: learn/using-contracts/setup.md (hunk 5 above)
Comment (ethanfrey) [reply learn/using-contracts/setup.md:30]: Nice idea. We should make this more extensible somehow.

There was already the request to upload all json schemas (and maybe all source) along with the wasm.
A URL for some cosmjs helpers is a good idea.

What do you think @webmaster128 ?

File: learn/using-contracts/cw20.md (hunk 6 above)
Comment (ethanfrey) [reply learn/using-contracts/cw20.md:75]: Ah, cool. Happy for any design improvements.

I would merge this PR once I address the comments, then let you add another PR to make it cleaner.

File: learn/using-contracts/setup.md (hunk 5 above)
Comment (webmaster128) [reply learn/using-contracts/setup.md:30]: Allowing untrusted "helpers" besically means contract uploaders can trick a user to execute any command they want on their local machine (such as uploading `~/.ssh/id_rsa` to a server or do nasty things with your metamask storage). 

What I could imagine having is an autogenerated helpers script that is generated by a trusted service like wasm.glass, which is a template with only few well validated variables.

Comment (ethanfrey) [reply learn/using-contracts/setup.md:30]: I just made an issue to discuss this more: https://github.com/CosmWasm/wasmd/issues/251

Feel free to comment there

**Comment by ethanfrey** [comment]
Hey @orkunkl thanks for the review.

As I was merging this in, I realize I didn't push my last commits (from 1 am), but just did now. This pulls out the `.type` stuff and more into an "advanced" directory, and fixes `token_info` => `tokenInfo`. Can you read through the new advanced usage section.

I do expect this to get refactored again as we add more contracts, but it would be great to have a read through

**Review by orkunkl — APPROVED** [review]
Good work! I liked the way the document is written. Grabs your attention.
